web/src/components/PreflightResultPage.jsx

```diff
@@ -314,6 +320,16 @@ class PreflightResultPage extends Component {
 
     const preflightSkipped = preflightResultData?.skipped;
     const stopPolling = (preflightResultData?.result || preflightSkipped);
+    let blockDeployment = false;
+    if (this.props.fromLicenseFlow) {
+      for (const app of this.props.appsList) {
+        if (app.slug == slug) {
+          blockDeployment = app.downstreams[0]?.pendingVersions[0]?.hasFailingStrictPreflights;
```

Comment:
why not just return this as part of the preflight results response? you already have the version sequence there. adding the "list app" job isn't needed.

Reply:
`hasFailingStrictPreflights`  is calculated from `app_version.preflight_spec` and `app_downstream_version.preflght_result`.
Will it be okay to joint these table with appID and sequence?

Reply:
added a new field `hasFailingStrictPreflights` to Preflight response and cleaned up the UI code.
Thanks for the suggestion